common/conf_test.cpp

```diff
@@ -337,21 +336,21 @@ TEST(ConfTests, LogLevel) {
 	{
 		vector<string> args {"--log-level", "error"};
 		conf::MenderConfig config;
-		config.ProcessCmdlineArgs(args.begin(), args.end(), cli::App {});
+		config.ProcessCmdlineArgs(args.begin(), args.end(), conf::CliApp {});
```

Comment:
Hmm, why not keep the `cli` namespace? :thinking:

Reply:
I wrote it in the commit message.

Originally I wanted to keep both independent, but it turns out that `cli` depends on `conf` to get the default paths and `conf` depends on `cli` to print the help pages on explicit `--help` argument.

On the other hand it was relatively little code in `cli` and also `conf` had already functionality around cmd line parsing... so I decided to merge (it is a separate commit so can be dropped if you push for keeping them separated).

Reply:
I decided to keep them together, and I did yet another refactoring for `ProcessCmdlineArgs` to get the flags from the commands structs instead of having the strings duplicated.